new_functions.py

```diff
@@ -135,54 +184,88 @@ def extrapolate_Y(y_vector: np.ndarray, length_of_padded_data: int) -> np.ndarra
     return np.concatenate([padding_start, y_vector, padding_end])
 
 
+def rescale_to_zero_centered_unit_range(
+    values: np.ndarray, min_value: float, max_value: float
+):
+    """
+    Rescale values to [-0.5, 0.5] range consistent with provided global min and max values
+
+    :param values: values to rescale
+    :param min_value: minimum of original range
+    :param max_value: maximum of original range
+
+    :return: Values rescaled to range of [-0.5, 0.5]
+    """
+    return (values - min_value) / (max_value - min_value) - 0.5
```

Comment:
```suggestion
    if min_value == max_value:
        raise ValueError("min_value and max_value must be different to avoid division by zero.")

    return (values - min_value) / (max_value - min_value) - 0.5
```

.. that said, not yet handling exceptions in the caller of this, but... Might as well prepare for it.